### PR TITLE
Edited heron_robot to work with uuv_simulator

### PR DESCRIPTION
**Pull request by guystoppi**

### CODE REVIEW

File: heron_base/launch/base.launch [hunk 1]
```diff
@@ -1,75 +1,82 @@
...
-
-  <!-- TODO: Eliminate this by compiling the necessary message definitions into heron_base. -->
-  <node pkg="rosserial_python" type="message_info_service.py" name="rosserial_message_info" />
-
-  <group if="$(optenv USE_SNMP_CHECK_NODE true)">
-    <node pkg="heron_base" type="snmp_lights.py" name="wifi_lights">
-	<param name="~ip_addr" value="$(optenv SNMP_IP 192.168.1.9)" />
+    <arg name="port" default="/dev/arduino" />
+    <node pkg="rosserial_server" type="serial_node" name="rosserial_server">
+      <param name="port" value="$(arg port)" />
```
Comment (tonybaltovski) [inline heron_base/launch/base.launch]: Is `<remap from="has_wifi" to="wireless/connected" />` missing?

File: heron_base/package.xml [hunk 2]
```diff
@@ -22,15 +22,14 @@
...
-  <run_depend>enu</run_depend>
   <run_depend>rosserial_python</run_depend>
   <run_depend>rosserial_server</run_depend>
   <run_depend>tf</run_depend>
   <run_depend>um6</run_depend>
   <run_depend>imu_compass</run_depend>
-  <run_depend>declination</run_depend>
   <run_depend>heron_control</run_depend>
   <run_depend>wireless_watcher</run_depend>
+  <run_depend>imu_filter_madgwick</run_depend>
```
Comment (tonybaltovski) [inline heron_base/package.xml]: Extra space

File: heron_base/launch/base.launch [hunk 3]
```diff
@@ -1,75 +1,101 @@
...
+    </group>
+
+    <group unless="$(eval arg('namespace') == '')">
+     <node pkg="um6" type="um6_driver" name="um6_driver" output="screen">
+       <param name="tf_ned_to_enu" value="true"/>
+       <param name="/um6_driver/port" value="$(optenv HERON_IMU_UM6_PORT /dev/clearpath/imu)" />
+       <remap from="imu/data" to="imu/data_raw" />
+       <param name="lt_model" value="true" />
+       <param name="linear_acceleration_stdev" value="0.38" />
+       <param name="frame_id" value="$(arg namespace)/imu_link" />
```
Comment (tonybaltovski) [inline heron_base/launch/base.launch:47]: Could thus just be an arg, instead of the grouping?

File: heron_nmea/REFERENCE.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+There are also a number of NMEA sentences published directly from the GPS sensor. The sentence type is prefixed by "GP" and can be decoded using this guide: http://www.gpsinformation.org/dale/nmea.htm
+
+#### Details of Output Data
+- CPIMU:
+  - Y Angular Velocity and Z Angular Velocity are negative compared to the direct output from the IMU (in degrees)
+  - Similarly for Y,Z linear acceleration
+- CPRCM:
+  - Heading field is (180 - deg(h)) where h is the heading from the IMU
+    - Heading field is zero when facing south and increases counter-clockwise
```
Comment (tonybaltovski) [inline heron_nmea/REFERENCE.md]: Is this supposed to be on a different level?

**Review by tonybaltovski — CHANGES_REQUESTED** [review]
Could we split out the controller changes from `heron_nema` for now?

File: heron_base/launch/base.launch (hunk 3 above)
Comment (guystoppi) [reply heron_base/launch/base.launch:47]: Do you mean a namespace to replace the "imu_link" transform frame? Or another way of using the namespace envar?

Comment (tonybaltovski) [reply heron_base/launch/base.launch:47]: Could it just be `"$(arg namespace)/imu_link"` or does that give double `/`?

Comment (guystoppi) [reply heron_base/launch/base.launch:47]: I'm not sure if that gives a double ```/``` or not, but even a single ```/``` is problematic. It results in:

``` Error was Invalid argument "/imu_link" passed to lookupTransform argument source_frame in tf2 frame_ids cannot start with a '/' like..."```

Comment (tonybaltovski) [reply heron_base/launch/base.launch:47]: Ah yes, you are correct!

**Review by dniewinski — APPROVED** [review]

**Review by tonybaltovski — APPROVED** [review]

**Pull request merged**